apps/preferences.py

```diff
@@ -421,11 +421,13 @@ def assignment_notification_is_enabled(self, user_id=None, preferences=None):
         send_email = preferences.get("assignment:notification", {}) if isinstance(preferences, dict) else {}
         return send_email and send_email.get("enabled", False)
 
-    def mark_for_user_notification_is_enabled(self, user_id=None, preferences=None):
+    def check_notification_is_enabled(self, preference_name, user_id=None, preferences=None):
         """
         This function checks if email notification is enabled or not based on the preferences.
         """
+        send_email = {}
         if user_id:
             preferences = self.get_user_preference(user_id)
-        send_email = preferences.get("mark_for_user:notification", {}) if isinstance(preferences, dict) else {}
+        if preference_name:
+            send_email = preferences.get(f"{preference_name}:notification", {}) if isinstance(preferences, dict) else {}
```

Comment:
imo we should have a specific `email_notification` later we might have separate in app notification setting, it's not clear which one is that

Reply:
this one is for mark_for_user mails, it's a child of `"email:notification" `, if the parent `email:notification` is disabled, none of email is send

Reply:
I'm not sure how that logic would work if we make it configurable for app notifications, so imo there should be some indication this is about email. if we have some concept of parent child, it should be more obvious imo.
I suppose `desktop:notification` is not child of `email:notification`. maybe it should be then `mark_for_user:email:notification` or `email:notification:mark_for_user` if we want to have some hierarchy

Reply:
makes sense